jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/params/GeneralParams.scala

```diff
@@ -81,16 +81,17 @@ trait GeneralParams extends Params {
     * The hdfs folder to load and save checkpoint boosters. default: `empty_string`
     */
   val checkpointPath = new Param[String](this, "checkpoint_path", "the hdfs folder to load and " +
-    "save checkpoints. The job will try to load the existing booster as the starting point for " +
-    "training. If saving_frequency is also set, the job will save a checkpoint every a few rounds.")
+    "save checkpoints. If there are existing checkpoints in checkpoint_path. The job will load " +
+    "the checkpoint with highest version as the starting point for training. If " +
+    "checkpoint_interval is also set, the job will save a checkpoint every a few rounds.")
 
   /**
-    * The frequency to save checkpoint boosters. default: 0
+    * The interval to save checkpoints. default: 0
     */
-  val savingFrequency = new IntParam(this, "saving_frequency", "if checkpoint_path is also set," +
-    " the job will save checkpoints at this frequency. If the job fails and gets restarted with" +
-    " same setting, it will load the existing booster instead of training from scratch." +
-    " Checkpoint will be disabled if set to 0.")
+  val checkpointInterval = new IntParam(this, "checkpoint_interval", "if checkpoint_path is also" +
```

Comment:
See above comment on Duration/TimeUnit. Alternatively, document the unit and add >=0 validator.

Reply:
I changed it to be align with the doc for checkpointInterval in Spark: https://github.com/apache/spark/blob/master/mllib/src/main/scala/org/apache/spark/ml/param/shared/sharedParams.scala#L288

Does it make more sense now?